src/components/Contacts/ContactDetails/ContactDetailsTab/People/Items/EditContactOtherModal/EditContactOtherModal.test.tsx

```diff
@@ -0,0 +1,240 @@
+            <EditContactOtherModal
+              accountListId={accountListId}
+              isOpen={true}
+              handleClose={handleClose}
+              contact={mockContact}
+            />
+          </GqlMockedProvider>
+        </ThemeProvider>
+      </SnackbarProvider>,
+    );
+
+    expect(getByText('Edit Contact Other Details')).toBeInTheDocument();
+    userEvent.click(getByRole('button', { name: 'Close' }));
+    expect(handleClose).toHaveBeenCalled();
+  });
+
+  it('should handle cancel click', () => {
+    const { getByText } = render(
+      <SnackbarProvider>
+        <ThemeProvider theme={theme}>
+          <GqlMockedProvider<UpdateContactOtherMutation>>
+            <EditContactOtherModal
+              accountListId={accountListId}
+              isOpen={true}
+              handleClose={handleClose}
+              contact={mockContact}
+            />
+          </GqlMockedProvider>
+        </ThemeProvider>
+      </SnackbarProvider>,
+    );
+
+    expect(getByText('Edit Contact Other Details')).toBeInTheDocument();
+    userEvent.click(getByText('Cancel'));
+    expect(handleClose).toHaveBeenCalled();
+  });
+
+  it('should edit contact other details', async () => {
+    const mutationSpy = jest.fn();
+    const newChurchName = 'Great Cool Church II';
+    const newWebsite = 'coolwebsite2.com';
+    const { getByText, getByRole } = render(
+      <SnackbarProvider>
+        <ThemeProvider theme={theme}>
+          <GqlMockedProvider<UpdateContactOtherMutation>
+            onCall={mutationSpy}
+            mocks={{
+              LoadConstants: {
+                constant: {
+                  languages: [
+                    {
+                      id: 'en',
+                      value: 'English',
+                    },
+                    {
+                      id: 'elx',
+                      value: 'Greek',
+                    },
+                    {
+                      id: 'eka',
+                      value: 'Ekajuk',
+                    },
+                    {
+                      id: 'en-AU',
+                      value: 'Australian English',
+                    },
+                  ],
+                },
+              },
+            }}
+          >
+            <EditContactOtherModal
+              accountListId={accountListId}
+              isOpen={true}
+              handleClose={handleClose}
+              contact={mockContact}
+            />
+          </GqlMockedProvider>
+        </ThemeProvider>
+      </SnackbarProvider>,
+    );
+
+    userEvent.clear(getByRole('textbox', { name: 'Church' }));
+    userEvent.clear(getByRole('textbox', { name: 'Website' }));
+
+    userEvent.click(getByRole('button', { name: 'Preferred Contact Method' }));
+    userEvent.click(getByRole('option', { name: 'WhatsApp' }));
+    // userEvent.click(getByRole('button', { name: 'Language' }));
+    // userEvent.click(getByRole('option', { name: 'Australian English' }));
```

Comment:
For some reason, I couldn't get the language selection to work. My mocks are working and I can see the data when debugging so not sure why this wasn't working.